proxy/InkAPI.cc

```diff
@@ -7647,7 +7647,11 @@ TSHttpTxnServerPush(TSHttpTxn txnp, const char *url, int url_len)
   HttpSM *sm          = reinterpret_cast<HttpSM *>(txnp);
   Http2Stream *stream = dynamic_cast<Http2Stream *>(sm->ua_session);
   if (stream) {
-    stream->push_promise(url_obj);
+    Http2ClientSession *parent = static_cast<Http2ClientSession *>(stream->get_parent());
+    if (!parent->is_url_pushed(url)) {
```

Comment:
We can't assume that `url` is null-terminated.

This check and the hash table should probably be in `Http2ConnectionState` so we don't need to add this check here and there.